flink-clients/src/main/java/org/apache/flink/client/deployment/application/ApplicationClusterEntryPoint.java

```diff
@@ -114,4 +115,16 @@ protected static void configureExecution(
         return Collections.unmodifiableList(
                 classpath.stream().distinct().collect(Collectors.toList()));
     }
+
+    @Override
+    public void cleanupDirectories() throws IOException {
+        try {
+            // Close the packaged program explicitly to clean up temporary jars.
+            program.close();
```

Comment:
this will never throw an exception, and even if then we wouldn't want to rethrow it because it would skip `super.cleanupDirectories`.